### ISSUE TITLE
Files in .vscode and .che folders of a git repository are ignored at workspace creation

### ISSUE THREAD

**Issue by svor** [eclipse/che#21635]
### Describe the bug

All defined plugins from the extensions.json are not installed If create a workspace by the factory and **che-theia** editor. 
Also I see that the tools component doesn't contain environment variables that should be injected by the Dashboard:
```
        - name: PLUGIN_REMOTE_ENDPOINT_EXECUTABLE
          value: /remote-endpoint/plugin-remote-endpoint
        - name: THEIA_PLUGINS
          value: 'local-dir:///plugins/sidecars/tools'
```

It's also not possible to customize an editor by using `.che/che-editor.yaml` file:

![screenshot-nimbusweb me-2022 09 07-10_02_22](https://user-images.githubusercontent.com/1271546/188815436-9855aa6d-f336-4495-a3d0-692b59dc191a.png)

### Che version

next (development version)

### Steps to reproduce

1. Create a workspace by this factory: https://github.com/che-samples/java-spring-petclinic/tree/devfilev2?che-editor=eclipse/che-theia/next
2. Open tools container and check that `/plugins` folder doesn't exist
3. Go to the Dasboard and check env variables in the tools component

### Expected behavior

`/plugins` folder should be existed and plugins from the .vscode/extensions.json should be installed

### Runtime

OpenShift

### Screenshots

![screenshot-che-dogfooding apps che-dev x6e0 p1 openshiftapps com-2022 08 15-16_52_45](https://user-images.githubusercontent.com/1271546/184648473-5bf2f058-65f0-4ca7-9f4a-cd20e8d96e24.png)

### Installation method

chectl/next

### Environment

Linux

### Release Notes Text

Files in `.vscode` and `.che` subfolders need to be processed during workspace creation as they can contain important workspace details (as the editor and the che-theia plugins to pre-install) but were ignored in recent versions due to a regression. This has been fixed in this release.

**Comment by svor** [comment]
cc @olexii4 @akurinnoy

**Comment by olexii4** [comment]
I think it happened after this [PR](https://github.com/eclipse-che/che-theia/pull/1361).
@vitaliy-guliy WDYT?

**Comment by vitaliy-guliy** [comment]
> I think it happened after this [PR](https://github.com/eclipse-che/che-theia/pull/1361) into the repo https://github.com/eclipse-che/che-theia. @vitaliy-guliy WDYT?

Due to all the tests in [devworkspace-handler](https://github.com/eclipse-che/che-theia/tree/main/tools/devworkspace-handler) are passed, I think the problem might be something else.
I will check it anyway, give me some time.

**Comment by vitaliy-guliy** [comment]
It looks like I found cause of the problem.
First, when create a workspace from provided samples on the dashboard, everything is working well.

When the user create a workspace from git repository, dashboard tries to fetch `.vscode/extensions.json` file to get the list of extensions for the following adding them to the devworkspace object.
But for some reasons, GET request on `.vscode/extensions.json` fails.

![Screenshot from 2022-09-02 18-52-50](https://user-images.githubusercontent.com/1655894/188612695-12894ab8-d353-496d-b6c7-c17bc4cab66e.png)

It becomes much clear when trying to open the URL in the new tab
![Screenshot from 2022-09-06 12-32-38](https://user-images.githubusercontent.com/1655894/188612959-63a29128-57cd-4355-bf7a-134c10eed611.png)

I tried to get other files from git repository and was able to get file content when the file or directory name do not contain dot at the beginning.

There is a potential place in che-server to fix the bug
https://github.com/eclipse-che/che-server/blob/main/wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/scm/AuthorizingFileContentProvider.java#L149
